plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraClusteringPredicatesExtractor.java

```diff
@@ -65,102 +67,135 @@ private static ClusteringPushDownResult getClusteringKeysSet(List<CassandraColum
             if (domain.isNullAllowed()) {
                 break;
             }
+
+            int currentlyProcessedClusteringColumn = allProcesedClusteringColumns;
             String predicateString = domain.getValues().getValuesProcessor().transform(
                     ranges -> {
-                        List<Object> singleValues = new ArrayList<>();
-                        List<String> rangeConjuncts = new ArrayList<>();
-                        String predicate = null;
+                        if (ranges.getRangeCount() == 1) {
+                            fullyPushedDomains.add(columnHandle);
+                            return translateRangeIntoCql(columnHandle, getOnlyElement(ranges.getOrderedRanges()));
+                        }
+                        if (ranges.getOrderedRanges().stream().allMatch(Range::isSingleValue)) {
+                            if (isInStatementNotAllowed(clusteringColumns, cassandraVersion, currentlyProcessedClusteringColumn)) {
+                                return translateRangeIntoCql(columnHandle, ranges.getSpan());
```

Comment:
Some of the cases in `translateRangeIntoCql` perform full pushdown so should the column handle be added to `fullyPushedDomains` too?